packages/react-dom-bindings/src/client/ReactDOMFloatClient.js

```diff
@@ -795,8 +969,104 @@ function createPreloadResource(
   };
 }
 
+function acquireHeadResource(resource: HeadResource): Instance {
+  resource.count++;
+  let instance = resource.instance;
+  if (!instance) {
+    const {props, root, type} = resource;
+    switch (type) {
+      case 'title': {
+        const titles = root.querySelectorAll('title');
+        for (let i = 0; i < titles.length; i++) {
+          if (titles[i].textContent === props.children) {
+            instance = resource.instance = titles[i];
+            markNodeAsResource(instance);
+            return instance;
+          }
+        }
+        instance = resource.instance = createResourceInstance(
+          type,
+          props,
+          root,
+        );
+        insertResourceInstanceBefore(root, instance, titles.item(0));
```

Comment:
Why `.item(0)` instead of `[0]` like you do above?

Reply:
Item returns null when index overflowing and array notation returns undefined. The insertBefore api technically requires a Node or null though I suspect it would also work with undefined. But I figured it was better to be technically correct